### PR TITLE
VRT vault

### PR DESCRIPTION
**Pull request by defistar**
## Description

- VRTVault contracts to let user deposit VRT for a fixed Interest Rate
- No LockTime for the funds
- User can deposit as many time as they wish to
- Interest is accrued based on blocks passed from the time since deposit was made
- Any existing deposit gets interest accrued and transferred to the User
- InterestRate is immutable and set once during contract deployment

## VRTVault Public and external functions
1. Deposit
2. Withdraw
3. Claim
4. getAccruedInterest
5. getBlockNumber

## VRTVault Admin functions

1. getAdmin
2. burnAdmin
3. setNewAdmin
4. setVenusInfo
5. _become
6. _setPendingImplementation
7. _acceptImplementation
8. _setPendingAdmin
9. _acceptAdmin

### CODE REVIEW

**Review by kkirka — CHANGES_REQUESTED** [review]

File: contracts/VRTVault/VRTVault.sol
Comment (defistar) [reply contracts/VRTVault/VRTVault.sol:87]: @kkirka will change this. to

                vrt.safeTransfer(user.userAddress, accruedInterest);

File: contracts/VRT/VRT.sol
Comment (defistar) [reply contracts/VRT/VRT.sol]: Hi @kkirka 
i have refactored `VRT` to contracts/Tokens

shall i refactor XVS and SXP in this PR?
or
shall i make it as a new PR?

Comment (kkirka) [reply contracts/VRT/VRT.sol]: Definitely not in this PR, let's have a new one. Please create a new task for this and schedule it to the next sprint.

Comment (defistar) [reply contracts/VRT/VRT.sol]: @kkirka  I made changes only for VRT 
  - move VRT to directory Tokens
  - VRT will refer to Tokenlock and Owned contracts  in Utils directory
  
shall i retain them or revert?

** VRT contract never exist in contracts repo, i got its code from mainnet, and created the files as-is

File: contracts/VRTVault/VRTVault.sol
Comment (kkirka) [inline contracts/VRTVault/VRTVault.sol:52]: No need to check msg.sender for zero

File: tests/VRTVault/vrtVaultTest.js
Comment (kkirka) [inline tests/VRTVault/vrtVaultTest.js:222]: The rest of the project uses camelCase without underscores for variable names 🤔

**Review by kkirka — CHANGES_REQUESTED** [review]

File: contracts/VRTVault/VRTVault.sol
Comment (kkirka) [inline contracts/VRTVault/VRTVault.sol:219]: We can use proxy functions to get and set admins, do we really need them in the proxy? `setNewAdmin` removes two-step admin change, and emits an event different from the one that proxy emits.

File: contracts/VRTVault/VRTVault.sol
Comment (kkirka) [inline contracts/VRTVault/VRTVault.sol:232]: We set these in `initialize`, looks like we don't need this function

File: contracts/VRTVault/VRTVault.sol
Comment (kkirka) [inline contracts/VRTVault/VRTVault.sol:31]: Why do we need the constructor at all here? To simplify testing?

File: contracts/VRTVault/VRTVaultStorage.sol
Comment (kkirka) [inline contracts/VRTVault/VRTVaultStorage.sol:19]: bscscan does not recognize `vrtVaultImplementation` as a pointer to the implementation contract. Let's use a conventional name here:

```suggestion
    address public implementation;
```

File: contracts/VRTVault/VRTVaultStorage.sol
Comment (kkirka) [inline contracts/VRTVault/VRTVaultStorage.sol:24]: And here as well:

```suggestion
    address public pendingImplementation;
```

File: contracts/VRTVault/VRTVault.sol
Comment (defistar) [reply contracts/VRTVault/VRTVault.sol:219]: @kkirka yeah unused.  these were the functions added from XVSVault / VAIVault contracts
i will remove these.

File: contracts/VRTVault/VRTVault.sol
Comment (defistar) [reply contracts/VRTVault/VRTVault.sol:31]: @kkirka for testing

File: contracts/VRTVault/VRTVault.sol
Comment (defistar) [reply contracts/VRTVault/VRTVault.sol:232]: Removed the function

File: contracts/VRTVault/VRTVaultStorage.sol
Comment (defistar) [reply contracts/VRTVault/VRTVaultStorage.sol:19]: renamed to implementation

File: contracts/VRTVault/VRTVaultStorage.sol
Comment (defistar) [reply contracts/VRTVault/VRTVaultStorage.sol:24]: renamed to pendingImplementation

**Comment by kkirka** [comment]
Task linked: [CU-22h3eka VRT vault](https://app.clickup.com/t/22h3eka)

**Review by kkirka — APPROVED** [review]

**Comment by kkirka** [comment]
Please rebase --autosquash before merging, and wait for @RoseRompkxm review as well

**Review by RoseRompkxm — APPROVED** [review]

**Pull request merged**
